pkg/standalone-utils/contracts/ERC4626CowSwapFeeBurner.sol

```diff
@@ -0,0 +1,107 @@
+
+    modifier onlyProtocolFeeSweeper() {
+        if (msg.sender != address(protocolFeeSweeper)) {
+            revert SenderNotAllowed();
+        }
+        _;
+    }
+
+    constructor(
+        IProtocolFeeSweeper _protocolFeeSweeper,
+        IVault vault,
+        IComposableCow composableCow,
+        address vaultRelayer,
+        bytes32 appData,
+        string memory version
+    ) CowSwapFeeBurner(vault, composableCow, vaultRelayer, appData, version) {
+        if (address(_protocolFeeSweeper) == address(0)) {
+            revert InvalidProtocolFeeSweeper();
+        }
+
+        protocolFeeSweeper = _protocolFeeSweeper;
+    }
+
+    /**
+     * @notice Treats `feeToken` as an ERC4626, redeems `exactFeeTokenAmountIn`, swaps the underlying asset for
+     * `targetToken`, and send proceeds to the `recipient`.
+     * @dev Assumes the sweeper has granted allowance for the fee tokens to the burner prior to the call.
+     * @param pool The pool the fees came from (only used for documentation in the event)
+     * @param feeToken The token collected from the pool
+     * @param exactFeeTokenAmountIn The number of fee tokens collected
+     * @param targetToken The desired target token (token out of the swap)
+     * @param minTargetTokenAmountOut The minimum amount out for the swap
+     * @param recipient The recipient of the swap proceeds
+     * @param deadline Deadline for the burn operation (i.e., swap), after which it will revert
+     */
+    function burn(
+        address pool,
+        IERC20 feeToken,
+        uint256 exactFeeTokenAmountIn,
+        IERC20 targetToken,
+        uint256 minTargetTokenAmountOut,
+        address recipient,
+        uint256 deadline
+    ) external override onlyProtocolFeeSweeper nonReentrant {
+        // In this case we first pull the wrapped token, unwrap, and then proceed to burn by creating an order for
+        // the underlying token.
+        feeToken.safeTransferFrom(msg.sender, address(this), exactFeeTokenAmountIn);
+
+        // Redeem and overwrite inputs with new asset and unwrapped amount.
+        IERC4626 erc4626Token = IERC4626(address(feeToken));
+        feeToken = IERC20(erc4626Token.asset());
+        exactFeeTokenAmountIn = erc4626Token.redeem(exactFeeTokenAmountIn, address(this), address(this));
+
+        // This case is not handled by the internal `_burn` function, but it's valid: we can consider that the token
+        // has already been converted to the correct token, so we just forward the result and finish.
+        if (feeToken == targetToken) {
+            // We apply the slippage check, but not deadline as the order settlement is instant in this case.
+            if (exactFeeTokenAmountIn < minTargetTokenAmountOut) {
+                revert AmountOutBelowMin(targetToken, exactFeeTokenAmountIn, minTargetTokenAmountOut);
+            }
+
+            feeToken.safeTransfer(recipient, exactFeeTokenAmountIn);
+            return;
```

Comment:
Consider if/else vs. early return, since there's only one line after it.

Reply:
that's fair